netpyne/cell/cell.py

```diff
@@ -197,7 +197,11 @@ def recordTraces (self):
                             ptr = getattr(getattr(self.secs[params['sec']]['hObj'](params['loc']), params['mech']), '_ref_'+params['var'])
                         elif 'synMech' in params:  # eg. soma(0.5).AMPA._ref_g
                             sec = self.secs[params['sec']]
-                            synMech = next((synMech for synMech in sec['synMechs'] if synMech['label']==params['synMech'] and synMech['loc']==params['loc']), None)
+                            synMechList = [synMech for synMech in sec['synMechs'] if synMech['label']==params['synMech'] and synMech['loc']==params['loc']] # make list with this label/loc
+                            if 'index' in params:
+                                synMech = synMechList[params['index']]
+                            else:
+                                synMech = synMechList[0] # 0th one which would have been returned by next()
```

Comment:
@wwlytton if the list is empty this will result in error ... should make synMech=None in that case

Reply:
@wwlytton: "added a try: to avoid error" -- thanks.

can you have a 'try' without the 'except'? 

also, can have same error when checking synMechList[params['index']] if the list is empty or not long enough.